newIDE/app/src/VariablesList/VariablesList.js

```diff
@@ -231,306 +238,336 @@ const VariableRow = React.memo<VariableRowProps>(
       ? valueAsString.indexOf('\n') !== -1
       : false;
 
+    const getContainerYPosition = React.useCallback(() => {
+      if (containerRef.current) {
+        return containerRef.current.getBoundingClientRect().top;
+      }
+    }, []);
+
     return (
-      <DragSourceAndDropTarget
-        beginDrag={() => {
-          draggedNodeId.current = nodeId;
-          return {};
-        }}
-        canDrag={() => !isInherited}
-        canDrop={() => canDrop(nodeId)}
-        drop={() => {
-          dropNode(nodeId);
-        }}
-      >
-        {({ connectDragSource, connectDropTarget, isOver, canDrop }) =>
-          connectDropTarget(
-            <div
-              style={{
-                marginLeft: (isNarrow ? 16 : 32) * depth,
-                backgroundColor: isSelected
-                  ? gdevelopTheme.listItem.selectedBackgroundColor
-                  : gdevelopTheme.list.itemsBackgroundColor,
-                marginBottom: 1,
-              }}
-              aria-selected={isSelected}
-              aria-expanded={isExpanded}
-              onPointerUp={event => {
-                const shouldMultiSelect = event.metaKey || event.ctrlKey;
-                onSelect(shouldMultiSelect, nodeId);
-              }}
-            >
-              {isOver && <DropIndicator canDrop={canDrop} />}
+      <div ref={containerRef}>
+        <DragSourceAndDropTarget
+          beginDrag={() => {
+            draggedNodeId.current = nodeId;
+            return {};
+          }}
+          canDrag={() => !isInherited}
+          canDrop={() => canDrop(nodeId)}
+          drop={() => {
+            dropNode(nodeId, whereToDrop);
+          }}
+          hover={monitor => {
+            const { y } = monitor.getClientOffset();
+            // Use a cached version of container position to avoid recomputing bounding rectangle.
+            // Doing this, the position is computed every second the user hovers the target.
+            const containerYPosition = memoized(
+              Math.floor(Date.now() / 1000),
+              getContainerYPosition
+            );
+            if (containerYPosition) {
+              setWhereToDrop(y - containerYPosition <= 18 ? 'before' : 'after');
```

Comment:
what's this 18? the height of the container? Is it worth adding a comment, or calculating this value based on the actual size of the element? (if possible and worth it)

Reply:
Good idea, thanks!